index.js

```diff
@@ -198,39 +134,159 @@ function generateImage(layout, callback) {
 
 
 /**
- * Spritezero can generate 2 kinds of layout objects: {@link DataLayout} and {@link ImgLayout}.
+ * Generate a {Manifest} object with information on where the icons are on the sprite.
+ *
+ * @param  {Layout}   layout    An {@link Layout} Object used to generate the image
+ * @return {Manifest} layout    A Manifest object
+ */
+function generateManifest(layout) {
+    assert(typeof layout === 'object');
+    if (!layout.items.length) return {};
+
+    var obj = {};
+    layout.items.forEach(function(item) {
+        item.names.forEach(function(name) {
+            obj[name] = {
+                width: item.width,
+                height: item.height,
+                x: item.x,
+                y: item.y,
+                pixelRatio: layout.pixelRatio
+            };
+        });
+    });
+    return obj;
+}
+
+function encodeBinaryManifest(manifest) {
+    function writeImage(data, pbf) {
+        pbf.writeStringField(1, data.id);
+        pbf.writeVarintField(2, data.img.width);
+        pbf.writeVarintField(3, data.img.height);
```

Comment:
It's common to have `height === width`, so we could save a few bytes by not encoding height for square images.

Reply:
Yup, but we could implement all sorts of optimizations like delta encoding etc., but I opted for the simplest format. The metadata is pretty small encoded as PBF compared to the image data anyway.